src/base/vesting/vesting.ts

```diff
@@ -37,50 +46,88 @@ export async function withdrawVested(
   );
   const signer = wallet.signer;
 
-  state.set("withdrawingSign");
+  state.set({ type: "withdrawingSign" });
+
+  try {
+    const tx: TransactionResponse = await contract
+      .connect(signer)
+      .withdrawVested();
 
-  const tx = await contract.connect(signer).withdrawVested();
+    state.set({ type: "withdrawing", tx });
+    await tx.wait();
+  } catch (e: any) {
```

Comment:
We shouldn't cast errors to `any`, by default it's `unknown`. Further down we should check that the error actually has a `.reason` attribute.

Reply:
28746cd should improve it